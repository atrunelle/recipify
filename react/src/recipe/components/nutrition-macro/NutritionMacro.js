import React, { Component } from 'react';
import NutrientsMacroChart from '../nutrients-macro-chart/NutrientsMacroChart';
import { Card, CardContent, Typography } from '@material-ui/core';

class NutritionMacro extends Component {
  render() {
    if (!this.props.totalNutrients.length) return '';

    return (
      <Card>
        <CardContent>
          <Typography
            gutterBottom 
            variant="headline"
            component="h2">
            Macro in recipe
          </Typography>          
          <Typography
            gutterBottom 
            paragraph
            component="p">
              Based on 4 calories per gram of carbs and protein, and 9 calories per gram of fat.
          </Typography>
          <Typography
            gutterBottom 
            paragraph
            component="p">
            Those numbers are approximation only, as the number of calories per macro change per ingredient and depend on quality and other factor.
          </Typography>
          <NutrientsMacroChart nutritionData={this.props.totalNutrients}></NutrientsMacroChart>
        </CardContent>
      </Card>
    );
  }
}

export default NutritionMacro;