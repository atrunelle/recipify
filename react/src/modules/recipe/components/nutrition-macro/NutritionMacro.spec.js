import React from 'react';
import { shallow } from 'enzyme';
import NutritionMacro from './NutritionMacro';

describe('Component: NutritionMacro', () => {
  it('should not render if totalNutrients is not passed', () => {
    const component = shallow(<NutritionMacro />);
    expect(component.html()).toBeNull();
  });

  it('should match snapshot', () => {
    const mockTotalNutrients = [{
      totalQuantity: {
        label: 'Fat',
        quantity: 10,
        percentage: 12,
        unit: 'gr',
      },
    }];

    const component = shallow(
      <NutritionMacro totalNutrients={mockTotalNutrients}/>
    );
    expect(component.html()).toMatchSnapshot();
  });

});