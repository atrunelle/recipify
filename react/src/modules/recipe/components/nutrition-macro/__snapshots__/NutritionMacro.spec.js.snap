// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NutritionMacro should match snapshot 1`] = `"<div class=\\"MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1\\"><div class=\\"MuiCardContent-root-29\\"><h2 class=\\"MuiTypography-root-30 MuiTypography-headline-35 MuiTypography-gutterBottom-47\\">Macro in recipe</h2><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47 MuiTypography-paragraph-48\\">Based on 4 calories per gram of carbs and protein, and 9 calories per gram of fat.</p><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-47 MuiTypography-paragraph-48\\">Those numbers are approximation only, as the number of calories per macro change per ingredient and depend on quality and other factor.</p><div class=\\"nutrients-macro-chart\\"><div style=\\"width:12%\\" class=\\"bar-graph bar-graph--fat\\"><p>Fat 12%</p></div></div></div></div>"`;
